src/monty/dev.py

```diff
@@ -129,14 +131,23 @@ def wrapped(*args, **kwargs):
         return wrapped
 
     def deprecated_class_decorator(cls: Type) -> Type:
-        original_init = cls.__init__
+        # Modify __post_init__ for dataclass
+        if is_dataclass(cls) and hasattr(cls, "__post_init__"):
```

Comment:
nice work! 👍 

i think `and hasattr(cls, "__post_init__")` is redundant. all `dataclasses` have `__post_init__`

Reply:
Thanks for the comment :)

That was added to prevent `mypy` from complaining:
```
"type[DataclassInstance]" has no attribute "__post_init__"  [attr-defined]
```

Personally I would prefer to avoid `# ignore: ignore` tags wherever we can, what do you think?

Reply:
> If no `__init__()` method is generated, then `__post_init__()` will not automatically be called.

not called is different from not present. but if it helps `mypy`, let's keep it but with a comment like `# prevent mypy attr-defined error`

Reply:
Thanks for the input. In that case, perhaps we just use the ignore tag to make it self-explanatory.

Reply:
It seems there are indeed cases where `__post_init__` doesn't exist:
```python
from dataclasses import dataclass

@dataclass
class SimpleDataClass:
    x: int
    y: int

example_inst = SimpleDataClass(x=10, y=20)
print(hasattr(example_inst, "__post_init__"))  # False
```

Reply:
oh my bad, good thing you checked!

Reply:
All good. We all make mistakes, **and learn!** :)